external/fv3fit/fv3fit/emulation/thermobasis/loss.py

```diff
@@ -0,0 +1,151 @@
+import dataclasses
+from typing import Any, List, Tuple
+import tensorflow as tf
+from fv3fit.emulation.thermobasis.thermo import ThermoBasis
+from fv3fit.emulation.thermo import relative_humidity, specific_humidity_from_rh
+
+
+def rh_loss_info(truth_rh, pred_rh, level):
+    loss_rh = tf.reduce_mean(tf.losses.mean_squared_error(truth_rh, pred_rh))
+    return {
+        f"relative_humidity_mse/level_{level}": loss_rh.numpy() * (86400 / 900) ** 2
+    }
+
+
+def q_loss_info(truth_q, pred_q, level):
+    loss_q = tf.reduce_mean(tf.losses.mean_squared_error(truth_q, pred_q))
+    return {
+        f"loss/variable_3/level_{level}": loss_q.numpy() * (1000 * 86400 / 900) ** 2
+    }
+
+
+@dataclasses.dataclass
+class QVLoss:
+    """Loss function for predicting specific humidity at a single level
+    
+    Attributes:
+        level: the level to predict
+        scale: the typical order of the loss function
+    """
+
+    level: int
+    scale: float = 1.0
+
+    def loss(self, pred: tf.Tensor, out: ThermoBasis):
+        truth_q = select_level(out.q, self.level)
+        loss = tf.reduce_mean(tf.losses.mean_squared_error(truth_q, pred))
+
+        pred_rh = relative_humidity(
+            select_level(out.T, self.level), pred, select_level(out.rho, self.level),
+        )
+        truth_rh = select_level(out.rh, self.level)
+
+        return (
+            loss / self.scale,
+            {
+                **q_loss_info(truth_q, pred, self.level),
+                **rh_loss_info(truth_rh, pred_rh, self.level),
+            },
+        )
+
+
+@dataclasses.dataclass
+class RHLoss:
```

Comment:
If this is for prediction at one level, could you make this clear from the name, and underscore in the docstrings whether tensors are single-level or multi-level?

Reply:
See 1faee88d